interop/tapir/src/main/scala/caliban/interop/tapir/HttpInterpreter.scala

```diff
@@ -52,15 +64,43 @@ object HttpInterpreter {
       graphQLRequest: GraphQLRequest,
       serverRequest: ServerRequest
     )(implicit streamConstructor: StreamConstructor[BS]): ZIO[R, TapirResponse, CalibanResponse[BS]] =
-      interpreter.executeRequest(graphQLRequest).map(buildHttpResponse[E, BS])
+      ZIO.scoped[R](config *> interpreter.executeRequest(graphQLRequest).map(buildHttpResponse[E, BS]))
+
+    def intercept[R1](interceptor: Interceptor[R1, R]): HttpInterpreter[R1, E] =
+      HttpInterpreter.Intercepted(this, interceptor)
+
+    def interceptWithZIO[R1](
+      interceptor: ServerRequest => ZIO[R1, TapirResponse, Unit]
+    )(implicit tag: Tag[R1]): HttpInterpreter[R & R1, E] =
+      HttpInterpreter.Intercepted[R & R1, R, E](
+        this,
+        ZLayer.environment[R] ++ ZLayer(ZIO.serviceWithZIO[ServerRequest](interceptor))
+      )
+
+    def configure[R1](configurator: Configurator[R1])(implicit tag: Tag[R1]): HttpInterpreter[R & R1, E] =
+      copy[R & R1, E](config = config *> configurator)
   }
 
-  private case class Configured[R1, R, E](
+  private case class Intercepted[R1, R, E](
     interpreter: HttpInterpreter[R, E],
     layer: ZLayer[R1 & ServerRequest, TapirResponse, R]
   ) extends HttpInterpreter[R1, E] {
-    override def intercept[R2](interceptor: Interceptor[R2, R1]): HttpInterpreter[R2, E] =
-      Configured[R2, R, E](interpreter, ZLayer.makeSome[R2 & ServerRequest, R](interceptor, layer))
+
+    def intercept[R2](interceptor: Interceptor[R2, R1]): HttpInterpreter[R2, E] =
+      copy[R2, R, E](layer = ZLayer.makeSome[R2 & ServerRequest, R](interceptor, layer))
+
+    def interceptWithZIO[R2](
+      interceptor: ServerRequest => ZIO[R2, TapirResponse, Unit]
+    )(implicit tag: Tag[R2]): HttpInterpreter[R1 & R2, E] =
+      copy[R1 & R2, R & R2, E](layer =
+        layer ++ ZLayer.environment[R2] ++ ZLayer(ZIO.serviceWithZIO[ServerRequest](interceptor))
+      )
+
+    def configure[R2](configurator: Configurator[R2])(implicit tag: Tag[R2]): HttpInterpreter[R1 & R2, E] =
+      Intercepted[R1 & R2, R & R2, E](
+        interpreter.configure[R2](configurator),
+        layer ++ ZLayer.environment[R2]
```

Comment:
Using `ZLayer.makeSome` doesn't require a `Tag` and is faster, I think?

Reply:
I’ve tried everything I could think of to get rid of the `Tag` / `++` and I can’t seem to get it to work (same for `configure`). 

When using `ZLayer.makeSome` alongside `ZLayer.environment`, a circular dependency compilation error is raised if the layer output contains a union type.